### ISSUE TITLE
`no-immediate-mutation` and objects with ES5- syntax

### ISSUE THREAD

**Issue by zloirock** [sindresorhus/eslint-plugin-unicorn#2796]
In some cases, I have code like this:
```js
function fn(key /*, ... */ {
  var object = { /* ... */ };
  object[key] = something;
  // ...
}

// or
var object = { /* ... */ };
var key = getKey();
object[key] = something;
// ...
```
`no-immediate-mutation` will report an error since it's an immediate mutation of an object.

But since this code should be written in ES5 (or even less ES version) syntax, I can't use computed properties.

It could be good to add an option to this rule that will allow such cases.
